## COLUMN OPERATIONS

In [1]:
import pandas as pd
import matplotlib.pyplot as mat
import seaborn as sea

##### Importing the libraries pandas ,matplotlib.pyplot an seaborn from python, in order to perform column operations on the dataset

In [2]:
sample = pd.read_csv("F:/Abi/python/python data/BIKE kaggle.csv")

In [3]:
sample

name  selling_price  year seller_type  \
0               Royal Enfield Classic 350         175000  2019  Individual   
1                               Honda Dio          45000  2017  Individual   
2     Royal Enfield Classic Gunmetal Grey         150000  2018  Individual   
3       Yamaha Fazer FI V 2.0 [2016-2018]          65000  2015  Individual   
4                   Yamaha SZ [2013-2014]          20000  2011  Individual   
...                                   ...            ...   ...         ...   
1056                            Activa 3g          17000  2010  Individual   
1057                     Honda CB twister          16000  2012  Individual   
1058                   Bajaj Discover 125          15000  2013  Individual   
1059                       Honda CB Shine          12000  2009  Individual   
1060                     Bajaj Pulsar 150          10000  2008  Individual   

          owner  km_driven  ex_showroom_price quality  
0     1st owner        350                NaN    good  
1     1st owner       5650                NaN    good  
2     1st owner      12000           148114.0    good  
3     1st owner      23000            89643.0    good  
4     2nd owner      21000                NaN    good  
...         ...        ...                ...     ...  
1056  1st owner     500000            52000.0     bad  
1057  1st owner      33000            51000.0     bad  
1058  2nd owner      35000            57000.0     bad  
1059  1st owner      53000            58000.0     bad  
1060  1st owner      92233            75000.0     bad  

[1061 rows x 8 columns]

In [4]:
sample.columns

Index(['name', 'selling_price', 'year', 'seller_type', 'owner', 'km_driven',
       'ex_showroom_price', 'quality'],
      dtype='object')

##### Displaying the variable names in the dataset.

In [5]:
sample['total']=sample['selling_price']+sample['ex_showroom_price']

##### Added a new variable in the dataset by combining the variables selling_price and ex_showroom_price

In [6]:
sample

name  selling_price  year seller_type  \
0               Royal Enfield Classic 350         175000  2019  Individual   
1                               Honda Dio          45000  2017  Individual   
2     Royal Enfield Classic Gunmetal Grey         150000  2018  Individual   
3       Yamaha Fazer FI V 2.0 [2016-2018]          65000  2015  Individual   
4                   Yamaha SZ [2013-2014]          20000  2011  Individual   
...                                   ...            ...   ...         ...   
1056                            Activa 3g          17000  2010  Individual   
1057                     Honda CB twister          16000  2012  Individual   
1058                   Bajaj Discover 125          15000  2013  Individual   
1059                       Honda CB Shine          12000  2009  Individual   
1060                     Bajaj Pulsar 150          10000  2008  Individual   

          owner  km_driven  ex_showroom_price quality     total  
0     1st owner        350                NaN    good       NaN  
1     1st owner       5650                NaN    good       NaN  
2     1st owner      12000           148114.0    good  298114.0  
3     1st owner      23000            89643.0    good  154643.0  
4     2nd owner      21000                NaN    good       NaN  
...         ...        ...                ...     ...       ...  
1056  1st owner     500000            52000.0     bad   69000.0  
1057  1st owner      33000            51000.0     bad   67000.0  
1058  2nd owner      35000            57000.0     bad   72000.0  
1059  1st owner      53000            58000.0     bad   70000.0  
1060  1st owner      92233            75000.0     bad   85000.0  

[1061 rows x 9 columns]

In [7]:
sample['low']=sample['selling_price']/sample['ex_showroom_price']

In [8]:
sample

name  selling_price  year seller_type  \
0               Royal Enfield Classic 350         175000  2019  Individual   
1                               Honda Dio          45000  2017  Individual   
2     Royal Enfield Classic Gunmetal Grey         150000  2018  Individual   
3       Yamaha Fazer FI V 2.0 [2016-2018]          65000  2015  Individual   
4                   Yamaha SZ [2013-2014]          20000  2011  Individual   
...                                   ...            ...   ...         ...   
1056                            Activa 3g          17000  2010  Individual   
1057                     Honda CB twister          16000  2012  Individual   
1058                   Bajaj Discover 125          15000  2013  Individual   
1059                       Honda CB Shine          12000  2009  Individual   
1060                     Bajaj Pulsar 150          10000  2008  Individual   

          owner  km_driven  ex_showroom_price quality     total       low  
0     1st owner        350                NaN    good       NaN       NaN  
1     1st owner       5650                NaN    good       NaN       NaN  
2     1st owner      12000           148114.0    good  298114.0  1.012733  
3     1st owner      23000            89643.0    good  154643.0  0.725098  
4     2nd owner      21000                NaN    good       NaN       NaN  
...         ...        ...                ...     ...       ...       ...  
1056  1st owner     500000            52000.0     bad   69000.0  0.326923  
1057  1st owner      33000            51000.0     bad   67000.0  0.313725  
1058  2nd owner      35000            57000.0     bad   72000.0  0.263158  
1059  1st owner      53000            58000.0     bad   70000.0  0.206897  
1060  1st owner      92233            75000.0     bad   85000.0  0.133333  

[1061 rows x 10 columns]

In [9]:
sample.rename(columns={'selling_price':'Sales'})

name   Sales  year seller_type  \
0               Royal Enfield Classic 350  175000  2019  Individual   
1                               Honda Dio   45000  2017  Individual   
2     Royal Enfield Classic Gunmetal Grey  150000  2018  Individual   
3       Yamaha Fazer FI V 2.0 [2016-2018]   65000  2015  Individual   
4                   Yamaha SZ [2013-2014]   20000  2011  Individual   
...                                   ...     ...   ...         ...   
1056                            Activa 3g   17000  2010  Individual   
1057                     Honda CB twister   16000  2012  Individual   
1058                   Bajaj Discover 125   15000  2013  Individual   
1059                       Honda CB Shine   12000  2009  Individual   
1060                     Bajaj Pulsar 150   10000  2008  Individual   

          owner  km_driven  ex_showroom_price quality     total       low  
0     1st owner        350                NaN    good       NaN       NaN  
1     1st owner       5650                NaN    good       NaN       NaN  
2     1st owner      12000           148114.0    good  298114.0  1.012733  
3     1st owner      23000            89643.0    good  154643.0  0.725098  
4     2nd owner      21000                NaN    good       NaN       NaN  
...         ...        ...                ...     ...       ...       ...  
1056  1st owner     500000            52000.0     bad   69000.0  0.326923  
1057  1st owner      33000            51000.0     bad   67000.0  0.313725  
1058  2nd owner      35000            57000.0     bad   72000.0  0.263158  
1059  1st owner      53000            58000.0     bad   70000.0  0.206897  
1060  1st owner      92233            75000.0     bad   85000.0  0.133333  

[1061 rows x 10 columns]

##### Renamed a 'selling_price' variablselling_price as 'Sales'

In [10]:
sample.rename(columns={'ex_showroom_price':'Original Price','total':'final'})

name  selling_price  year seller_type  \
0               Royal Enfield Classic 350         175000  2019  Individual   
1                               Honda Dio          45000  2017  Individual   
2     Royal Enfield Classic Gunmetal Grey         150000  2018  Individual   
3       Yamaha Fazer FI V 2.0 [2016-2018]          65000  2015  Individual   
4                   Yamaha SZ [2013-2014]          20000  2011  Individual   
...                                   ...            ...   ...         ...   
1056                            Activa 3g          17000  2010  Individual   
1057                     Honda CB twister          16000  2012  Individual   
1058                   Bajaj Discover 125          15000  2013  Individual   
1059                       Honda CB Shine          12000  2009  Individual   
1060                     Bajaj Pulsar 150          10000  2008  Individual   

          owner  km_driven  Original Price quality     final       low  
0     1st owner        350             NaN    good       NaN       NaN  
1     1st owner       5650             NaN    good       NaN       NaN  
2     1st owner      12000        148114.0    good  298114.0  1.012733  
3     1st owner      23000         89643.0    good  154643.0  0.725098  
4     2nd owner      21000             NaN    good       NaN       NaN  
...         ...        ...             ...     ...       ...       ...  
1056  1st owner     500000         52000.0     bad   69000.0  0.326923  
1057  1st owner      33000         51000.0     bad   67000.0  0.313725  
1058  2nd owner      35000         57000.0     bad   72000.0  0.263158  
1059  1st owner      53000         58000.0     bad   70000.0  0.206897  
1060  1st owner      92233         75000.0     bad   85000.0  0.133333  

[1061 rows x 10 columns]

##### Renamed Two variables 'ex_showroom_price' as 'Original Price' and 'total' as 'final'.

In [11]:
sample.drop(['seller_type','km_driven'],axis=1)

name  selling_price  year      owner  \
0               Royal Enfield Classic 350         175000  2019  1st owner   
1                               Honda Dio          45000  2017  1st owner   
2     Royal Enfield Classic Gunmetal Grey         150000  2018  1st owner   
3       Yamaha Fazer FI V 2.0 [2016-2018]          65000  2015  1st owner   
4                   Yamaha SZ [2013-2014]          20000  2011  2nd owner   
...                                   ...            ...   ...        ...   
1056                            Activa 3g          17000  2010  1st owner   
1057                     Honda CB twister          16000  2012  1st owner   
1058                   Bajaj Discover 125          15000  2013  2nd owner   
1059                       Honda CB Shine          12000  2009  1st owner   
1060                     Bajaj Pulsar 150          10000  2008  1st owner   

      ex_showroom_price quality     total       low  
0                   NaN    good       NaN       NaN  
1                   NaN    good       NaN       NaN  
2              148114.0    good  298114.0  1.012733  
3               89643.0    good  154643.0  0.725098  
4                   NaN    good       NaN       NaN  
...                 ...     ...       ...       ...  
1056            52000.0     bad   69000.0  0.326923  
1057            51000.0     bad   67000.0  0.313725  
1058            57000.0     bad   72000.0  0.263158  
1059            58000.0     bad   70000.0  0.206897  
1060            75000.0     bad   85000.0  0.133333  

[1061 rows x 8 columns]

##### We have dropped the variables 'seller_type' and 'km_driven' from the dataset

In [12]:
sample.columns=[col.upper()for col in sample.columns]

In [13]:
sample

NAME  SELLING_PRICE  YEAR SELLER_TYPE  \
0               Royal Enfield Classic 350         175000  2019  Individual   
1                               Honda Dio          45000  2017  Individual   
2     Royal Enfield Classic Gunmetal Grey         150000  2018  Individual   
3       Yamaha Fazer FI V 2.0 [2016-2018]          65000  2015  Individual   
4                   Yamaha SZ [2013-2014]          20000  2011  Individual   
...                                   ...            ...   ...         ...   
1056                            Activa 3g          17000  2010  Individual   
1057                     Honda CB twister          16000  2012  Individual   
1058                   Bajaj Discover 125          15000  2013  Individual   
1059                       Honda CB Shine          12000  2009  Individual   
1060                     Bajaj Pulsar 150          10000  2008  Individual   

          OWNER  KM_DRIVEN  EX_SHOWROOM_PRICE QUALITY     TOTAL       LOW  
0     1st owner        350                NaN    good       NaN       NaN  
1     1st owner       5650                NaN    good       NaN       NaN  
2     1st owner      12000           148114.0    good  298114.0  1.012733  
3     1st owner      23000            89643.0    good  154643.0  0.725098  
4     2nd owner      21000                NaN    good       NaN       NaN  
...         ...        ...                ...     ...       ...       ...  
1056  1st owner     500000            52000.0     bad   69000.0  0.326923  
1057  1st owner      33000            51000.0     bad   67000.0  0.313725  
1058  2nd owner      35000            57000.0     bad   72000.0  0.263158  
1059  1st owner      53000            58000.0     bad   70000.0  0.206897  
1060  1st owner      92233            75000.0     bad   85000.0  0.133333  

[1061 rows x 10 columns]

##### Displayed the Variables name of the dataset in the Uppercase.

In [14]:
sample.columns=[col.lower()for col in sample.columns]

In [15]:
sample

name  selling_price  year seller_type  \
0               Royal Enfield Classic 350         175000  2019  Individual   
1                               Honda Dio          45000  2017  Individual   
2     Royal Enfield Classic Gunmetal Grey         150000  2018  Individual   
3       Yamaha Fazer FI V 2.0 [2016-2018]          65000  2015  Individual   
4                   Yamaha SZ [2013-2014]          20000  2011  Individual   
...                                   ...            ...   ...         ...   
1056                            Activa 3g          17000  2010  Individual   
1057                     Honda CB twister          16000  2012  Individual   
1058                   Bajaj Discover 125          15000  2013  Individual   
1059                       Honda CB Shine          12000  2009  Individual   
1060                     Bajaj Pulsar 150          10000  2008  Individual   

          owner  km_driven  ex_showroom_price quality     total       low  
0     1st owner        350                NaN    good       NaN       NaN  
1     1st owner       5650                NaN    good       NaN       NaN  
2     1st owner      12000           148114.0    good  298114.0  1.012733  
3     1st owner      23000            89643.0    good  154643.0  0.725098  
4     2nd owner      21000                NaN    good       NaN       NaN  
...         ...        ...                ...     ...       ...       ...  
1056  1st owner     500000            52000.0     bad   69000.0  0.326923  
1057  1st owner      33000            51000.0     bad   67000.0  0.313725  
1058  2nd owner      35000            57000.0     bad   72000.0  0.263158  
1059  1st owner      53000            58000.0     bad   70000.0  0.206897  
1060  1st owner      92233            75000.0     bad   85000.0  0.133333  

[1061 rows x 10 columns]

##### Displayed the Variables name of the dataset in the lowercase

In [16]:
sample.sort_values(by=['selling_price','ex_showroom_price'],ascending=[True,False])

name  selling_price  year seller_type  \
543                       TVS Excel           5000  1998  Individual   
748        Honda Activa [2000-2015]           6000  2006  Individual   
567                     Bajaj Super           6000  1991  Individual   
77                        Honda Dio           7500  2009  Individual   
115         Hero Honda Passion Plus           7500  2004  Individual   
..                              ...            ...   ...         ...   
311      Harley-Davidson Street 750         330000  2014  Individual   
13                    Yamaha YZF R3         365000  2019  Individual   
289  Kawasaki Ninja 650 [2018-2019]         425000  2017  Individual   
134      Harley-Davidson Street Bob         750000  2013  Individual   
295                 Suzuki GSX S750         760000  2019  Individual   

         owner  km_driven  ex_showroom_price quality      total       low  
543  1st owner      60000                NaN    good        NaN       NaN  
748  2nd owner      35000            50267.0    good    56267.0  0.119363  
567  1st owner      35000                NaN    good        NaN       NaN  
77   2nd owner      16000                NaN     bad        NaN       NaN  
115  1st owner     120000                NaN     bad        NaN       NaN  
..         ...        ...                ...     ...        ...       ...  
311  4th owner       6500           534000.0    good   864000.0  0.617978  
13   1st owner       1127           351680.0    good   716680.0  1.037875  
289  2nd owner      13600           599000.0    good  1024000.0  0.709516  
134  2nd owner      12000          1278000.0     bad  2028000.0  0.586854  
295  1st owner       2800           752020.0    good  1512020.0  1.010611  

[1061 rows x 10 columns]

##### In this we have sorted two variables 'selling_price' and 'ex_showroom_price' and then we have arranged the variable 'selling_price' in the ascending order.

In [17]:
sample.sort_values(by='total',ascending=True)

name  selling_price  year seller_type      owner  \
1055              Bajaj  ct 100          18000  2017  Individual  1st owner   
748    Honda Activa [2000-2015]           6000  2006  Individual  2nd owner   
906                    Yo Style          20000  2011  Individual  1st owner   
456    Honda Activa [2000-2015]          10000  2009  Individual  1st owner   
851                  TVS XL 100          30000  2017  Individual  1st owner   
...                         ...            ...   ...         ...        ...   
951   Royal Enfield Electra 5 S          90000  2012  Individual  1st owner   
952             Hero Honda Hunk          20000  2010  Individual  1st owner   
957      Bajaj Pulsar 220 DTS-i          60000  2014  Individual  1st owner   
958      Hero Honda CBZ extreme          40000  2009  Individual  1st owner   
959      Hero Honda CBZ extreme          35000  2012  Individual  1st owner   

      km_driven  ex_showroom_price quality    total       low  
1055      35000            32000.0     bad  50000.0  0.562500  
748       35000            50267.0    good  56267.0  0.119363  
906       10000            37675.0    good  57675.0  0.530856  
456       60000            50267.0    good  60267.0  0.198938  
851        8000            30490.0    good  60490.0  0.983929  
...         ...                ...     ...      ...       ...  
951       40000                NaN    good      NaN       NaN  
952       17000                NaN    good      NaN       NaN  
957       16000                NaN    good      NaN       NaN  
958       50000                NaN    good      NaN       NaN  
959       60000                NaN    good      NaN       NaN  

[1061 rows x 10 columns]

##### Sorted the variable named 'total' and arranged its values in ascending order

In [18]:
sample.sort_values(by='total')

name  selling_price  year seller_type      owner  \
1055              Bajaj  ct 100          18000  2017  Individual  1st owner   
748    Honda Activa [2000-2015]           6000  2006  Individual  2nd owner   
906                    Yo Style          20000  2011  Individual  1st owner   
456    Honda Activa [2000-2015]          10000  2009  Individual  1st owner   
851                  TVS XL 100          30000  2017  Individual  1st owner   
...                         ...            ...   ...         ...        ...   
951   Royal Enfield Electra 5 S          90000  2012  Individual  1st owner   
952             Hero Honda Hunk          20000  2010  Individual  1st owner   
957      Bajaj Pulsar 220 DTS-i          60000  2014  Individual  1st owner   
958      Hero Honda CBZ extreme          40000  2009  Individual  1st owner   
959      Hero Honda CBZ extreme          35000  2012  Individual  1st owner   

      km_driven  ex_showroom_price quality    total       low  
1055      35000            32000.0     bad  50000.0  0.562500  
748       35000            50267.0    good  56267.0  0.119363  
906       10000            37675.0    good  57675.0  0.530856  
456       60000            50267.0    good  60267.0  0.198938  
851        8000            30490.0    good  60490.0  0.983929  
...         ...                ...     ...      ...       ...  
951       40000                NaN    good      NaN       NaN  
952       17000                NaN    good      NaN       NaN  
957       16000                NaN    good      NaN       NaN  
958       50000                NaN    good      NaN       NaN  
959       60000                NaN    good      NaN       NaN  

[1061 rows x 10 columns]

##### We have sorted the variable named 'total'.

In [19]:
sample.drop(['low'],axis=1,inplace=True)

##### We have dropped the variable named 'low' and used inplace function.


In [20]:
sample

name  selling_price  year seller_type  \
0               Royal Enfield Classic 350         175000  2019  Individual   
1                               Honda Dio          45000  2017  Individual   
2     Royal Enfield Classic Gunmetal Grey         150000  2018  Individual   
3       Yamaha Fazer FI V 2.0 [2016-2018]          65000  2015  Individual   
4                   Yamaha SZ [2013-2014]          20000  2011  Individual   
...                                   ...            ...   ...         ...   
1056                            Activa 3g          17000  2010  Individual   
1057                     Honda CB twister          16000  2012  Individual   
1058                   Bajaj Discover 125          15000  2013  Individual   
1059                       Honda CB Shine          12000  2009  Individual   
1060                     Bajaj Pulsar 150          10000  2008  Individual   

          owner  km_driven  ex_showroom_price quality     total  
0     1st owner        350                NaN    good       NaN  
1     1st owner       5650                NaN    good       NaN  
2     1st owner      12000           148114.0    good  298114.0  
3     1st owner      23000            89643.0    good  154643.0  
4     2nd owner      21000                NaN    good       NaN  
...         ...        ...                ...     ...       ...  
1056  1st owner     500000            52000.0     bad   69000.0  
1057  1st owner      33000            51000.0     bad   67000.0  
1058  2nd owner      35000            57000.0     bad   72000.0  
1059  1st owner      53000            58000.0     bad   70000.0  
1060  1st owner      92233            75000.0     bad   85000.0  

[1061 rows x 9 columns]

In [21]:
sample.groupby('selling_price').count()

name  year  seller_type  owner  km_driven  ex_showroom_price  \
selling_price                                                                 
5000              1     1            1      1          1                  0   
6000              2     2            2      2          2                  1   
7500              2     2            2      2          2                  0   
8000              3     3            3      3          3                  0   
9000              1     1            1      1          1                  0   
...             ...   ...          ...    ...        ...                ...   
330000            1     1            1      1          1                  1   
365000            1     1            1      1          1                  1   
425000            1     1            1      1          1                  1   
750000            1     1            1      1          1                  1   
760000            1     1            1      1          1                  1   

               quality  total  
selling_price                  
5000                 1      0  
6000                 2      1  
7500                 2      0  
8000                 3      0  
9000                 1      0  
...                ...    ...  
330000               1      1  
365000               1      1  
425000               1      1  
750000               1      1  
760000               1      1  

[130 rows x 8 columns]

##### By Performing groupby function on selling_price variable and used count function it is displays us this variable takes place in 130 observations.

In [22]:
sample = pd.read_csv("F:/Abi/python/python data/BIKE kaggle.csv")

In [23]:
sample

name  selling_price  year seller_type  \
0               Royal Enfield Classic 350         175000  2019  Individual   
1                               Honda Dio          45000  2017  Individual   
2     Royal Enfield Classic Gunmetal Grey         150000  2018  Individual   
3       Yamaha Fazer FI V 2.0 [2016-2018]          65000  2015  Individual   
4                   Yamaha SZ [2013-2014]          20000  2011  Individual   
...                                   ...            ...   ...         ...   
1056                            Activa 3g          17000  2010  Individual   
1057                     Honda CB twister          16000  2012  Individual   
1058                   Bajaj Discover 125          15000  2013  Individual   
1059                       Honda CB Shine          12000  2009  Individual   
1060                     Bajaj Pulsar 150          10000  2008  Individual   

          owner  km_driven  ex_showroom_price quality  
0     1st owner        350                NaN    good  
1     1st owner       5650                NaN    good  
2     1st owner      12000           148114.0    good  
3     1st owner      23000            89643.0    good  
4     2nd owner      21000                NaN    good  
...         ...        ...                ...     ...  
1056  1st owner     500000            52000.0     bad  
1057  1st owner      33000            51000.0     bad  
1058  2nd owner      35000            57000.0     bad  
1059  1st owner      53000            58000.0     bad  
1060  1st owner      92233            75000.0     bad  

[1061 rows x 8 columns]

In [24]:
print(sample['selling_price'].isnull().sum())

0


##### There is no  null values present in the 'selling_price' variable.

In [25]:
print(sample.columns.values)

['name' 'selling_price' 'year' 'seller_type' 'owner' 'km_driven'
 'ex_showroom_price' 'quality']


##### Displayed the variables but by using print function